python/ray/_raylet.pyx

```diff
@@ -1443,9 +1458,13 @@ cdef class CoreWorker:
             object_ref.native())
 
     def remove_object_ref_reference(self, ObjectRef object_ref):
-        # Note: faster to not release GIL for short-running op.
-        CCoreWorkerProcess.GetCoreWorker().RemoveLocalReference(
-            object_ref.native())
+        cdef:
+            CObjectID c_object_id = object_ref.native()
+        # We need to release the gil since object destruction may call the
+        # unhandled exception handler.
```

Comment:
If this is the only time that the unhandled exception handler is called, should we just hold the gil through the whole call to `RemoveLocalReference` and not acquire it in the handler?

Reply:
Sadly we need to hold it during the call to .native().